contracts/V1/StateChannelDiamondProxy/DisputeManagerFacet.sol

```diff
@@ -14,26 +14,28 @@ contract DisputeManagerFacet is StateChannelCommon {
         bytes memory signature
     ) public { 
         // sanity checks
-        msg.sender == dispute.disputer;
+        require(msg.sender == dispute.disputer, "CREATE DISPUTE: INVALID DISPUTER");
         // race condition check
-        if(onChainSlashes.length != onChainSlashedParticipants.length 
-        || onChainSlashes[onChainSlashes.length - 1] != onChainSlashedParticipants[onChainSlashedParticipants.length - 1]) {
-            revert("onChainSlashes does not match onChainSlashedParticipants");
+        address[] memory onChainSlashedParticipants = getOnChainSlashedParticipants();
+        if(dispute.onChainSlashes.length != onChainSlashedParticipants.length 
+        || dispute.onChainSlashes[dispute.onChainSlashes.length - 1] != onChainSlashedParticipants[onChainSlashedParticipants.length - 1]) {
+            revert CreateDisputeInvalidOnChainSlashedParticipants();
         }
 
-        address disputer = StateChannelUtilLibrary.retriveSignerAddress(dispute.encodedBlock, signature);
+        address disputer = StateChannelUtilLibrary.retriveSignerAddress(abi.encode(dispute), signature);
         if(disputer != dispute.disputer) {
-            revert("Invalid signature");
-        }
-        if(proofs.length == 0) return;
+            revert CreateDisputeInvalidSignature();
+        } 
+        // check if fraud proofs or TimoutStruct is provided to process the dispute
+        if(dispute.fraudProofs.length == 0 || isTimeoutSetWithOptional(dispute.timeout, false)) return;
+
         // commit to dispute struct
         bytes memory encodedDispute = abi.encode(dispute);
-        bytes32 memory channelIdHash = keccak256(dispute.channelId);
         bytes32 memory disputeCommitment = keccak256(abi.encodePacked(
```

Comment:
We shouldn't use abi.encodePacked - it's cheaper, but can have collision in hashing - in general we just use abi.encode everywhere since it's both collision safe for hashing and it can be easily decoded back and the gas savings don't make a difference